src/babel-serializer/network/pcap.hh

```diff
@@ -2,65 +2,116 @@
 
 #include <cstdint>
 
-#include <clean-core/array.hh>
-#include <clean-core/function_ref.hh>
 #include <clean-core/span.hh>
-#include <clean-core/vector.hh>
 
 #include <babel-serializer/errors.hh>
 
 namespace babel::pcap
 {
+namespace detail
+{
+struct packet_range; // fwd
+}
+
+/// This is the header for an entire pcap file
+struct header
+{
+    uint32_t magic_number;  ///< magic number
+    uint16_t version_major; ///< major version number
+    uint16_t version_minor; ///< minor version number
+    int32_t thiszone;       ///< GMT to local correction
+    uint32_t sigfigs;       ///< accuracy of timestamps
+    uint32_t snap_length;   ///< max length of captured packets, in octets
+    uint32_t network;       ///< data link type
+
+    /// returns true, if the endianness of all data must be swapped because the system that produced this
+    /// pcap file has opposite endianness than the system that is now reading the file
+    [[nodiscard]] constexpr bool swapped_endianness() const { return magic_number == 0xd4c3b2a1u || magic_number == 0x4d3cb2a1u; }
+
+    /// returns true, if the packet timestamps are nanoseconds instead of microseconds
+    [[nodiscard]] constexpr bool nanosecond_timestamps() const { return magic_number == 0xa1b23c4du || magic_number == 0x4d3cb2a1u; }
```

Comment:
it doesn't make sense to make these constexpr.
Also, I would just name them `has_...` and drop the nodiscard.

Reply:
Yeah you're right, constexpr does not make sense, when you most likely read your data from some file :+1:
I like the naming change! However, why would you drop the nodiscard?

Reply:
Discussed the last point externally, and applied the changes.